pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -230,51 +230,84 @@ func toRouteKey(c config.Config) RouteKey {
 func convertDestinationRule(r *KubernetesResources) []config.Config {
 	result := []config.Config{}
 	for _, obj := range r.BackendPolicy {
-		bp := obj.Spec.(*k8s.BackendPolicySpec)
+		res := convertBackendPolicy(r, obj)
+		result = append(result, res...)
+	}
+	return result
+}
 
-		// We have no status updates to write. The only condition defined is ConditionNoSuchBackend;
-		// since we translate to opaque host names we don't actually know
-		// TODO(https://github.com/kubernetes-sigs/gateway-api/issues/590) consider more fields in the API
+// TODO(https://github.com/kubernetes-sigs/gateway-api/issues/590) consider more fields in the API
+const BackendPolicyAdmitted = "Admitted"
 
-		for i, ref := range bp.BackendRefs {
-			var serviceName string
-			if emptyOrEqual(ref.Group, gvk.Service.CanonicalGroup()) && emptyOrEqual(ref.Kind, gvk.Service.Kind) {
-				serviceName = fmt.Sprintf("%s.%s.svc.%s", ref.Name, obj.Namespace, r.Domain)
-			} else {
-				log.Warnf("unsupported backendRef: %+v", ref)
-				continue
-			}
-			dr := &istio.DestinationRule{
-				Host:          serviceName,
-				TrafficPolicy: &istio.TrafficPolicy{},
-			}
-			if bp.TLS != nil && bp.TLS.CertificateAuthorityRef != nil {
-				tls := &istio.ClientTLSSettings{
-					// Currently, only simple is supported
-					CredentialName: buildSecretReference(*bp.TLS.CertificateAuthorityRef),
-					Mode:           istio.ClientTLSSettings_SIMPLE,
-				}
-				if ref.Port != nil {
-					dr.TrafficPolicy.PortLevelSettings = append(dr.TrafficPolicy.PortLevelSettings, &istio.TrafficPolicy_PortTrafficPolicy{
-						Port: &istio.PortSelector{Number: uint32(*ref.Port)},
-						Tls:  tls,
-					})
-				} else {
-					dr.TrafficPolicy.Tls = tls
-				}
+func convertBackendPolicy(r *KubernetesResources, obj config.Config) []config.Config {
+	result := []config.Config{}
+
+	bp := obj.Spec.(*k8s.BackendPolicySpec)
+
+	backendConditions := map[string]*condition{
+		BackendPolicyAdmitted: {
+			reason:  "Admitted",
+			message: "Configuration was valid",
+		},
+		// TODO actually look in push context to see service exists
+		string(k8s.ConditionNoSuchBackend): {
+			reason:  "BackendValid",
+			message: "All backends are valid",
+			status:  kstatus.StatusFalse,
+		},
+	}
+	defer reportBackendPolicyCondition(obj, backendConditions)
+
+	var tlsSettings *istio.ClientTLSSettings
+	if bp.TLS != nil && bp.TLS.CertificateAuthorityRef != nil {
+		cred, err := buildSecretReference(*bp.TLS.CertificateAuthorityRef)
+		if err != nil {
+			backendConditions[BackendPolicyAdmitted].error = err
+			return result
+		}
+		tlsSettings = &istio.ClientTLSSettings{
+			// Currently, only simple is supported
+			CredentialName: cred,
+			Mode:           istio.ClientTLSSettings_SIMPLE,
+		}
+	}
+	for i, ref := range bp.BackendRefs {
+		var serviceName string
+		if emptyOrEqual(ref.Group, gvk.Service.CanonicalGroup()) && emptyOrEqual(ref.Kind, gvk.Service.Kind) {
+			// TODO validate this actually exists in service registry?
```

Comment:
+1

Reply:
Will do it in https://github.com/istio/istio/pull/33306, or as a follow up to that one. That does it for Gateway, but not backendpolicy yet